tests/test_opds_import.py

```diff
@@ -416,50 +510,58 @@ def test_import_and_make_presentation_ready(self):
         importer = OPDSImporter(
             self._db, data_source_name=DataSource.OA_CONTENT_SERVER
         )
-        imported, messages, next_links = importer.import_from_feed(
-            feed, immediately_presentation_ready=True
+        imported_editions, imported_pools, imported_works, error_messages, next_link = (
+            importer.import_from_feed(feed, immediately_presentation_ready=True)
         )
 
-        [crow, mouse] = sorted(imported, key=lambda x: x.title)
+        [crow, mouse] = sorted(imported_editions, key=lambda x: x.title)
 
-        # Nothing happens for the 'crow' book.
-        eq_(None, crow.work)
-        
-        # But the 'mouse' book has had a presentation-ready work
-        # created for it.
+        # Both the 'crow' and the 'mouse' book had presentation-ready works created.
+        eq_(True, crow.license_pool.work.presentation_ready)
         eq_(True, mouse.license_pool.work.presentation_ready)
 
+
     def test_status_and_message(self):
         path = os.path.join(self.resource_path, "unrecognized_identifier.opds")
         feed = open(path).read()
-        imported, messages, next_link = OPDSImporter(self._db).import_from_feed(feed)
-        [message] = messages.values()
+        #imported, messages, next_link = OPDSImporter(self._db).import_from_feed(feed)
+        imported_editions, imported_pools, imported_works, error_messages, next_links = (
+            OPDSImporter(self._db).import_from_feed(feed)
+        )
+
+        [message] = error_messages.values()
         eq_(404, message.status_code)
         eq_("I've never heard of this work.", message.message)
 
+
     def test_import_failure_becomes_status_message(self):
+        # Make sure that an exception during import stops the import process, 
+        # and generates a meaningful error message.
 
         class DoomedOPDSImporter(OPDSImporter):
-            def import_from_metadata(self, metadata, *args):
+            def import_edition_from_metadata(self, metadata, *args):
                 if metadata.title == "Johnny Crow's Party":
                     # This import succeeds.
-                    return super(DoomedOPDSImporter, self).import_from_metadata(metadata, *args)
+                    return super(DoomedOPDSImporter, self).import_edition_from_metadata(metadata, *args)
                 else:
                     # Any other import fails.
                     raise Exception("Utter failure!")
         path = os.path.join(self.resource_path, "content_server_mini.opds")
         feed = open(path).read()
-        imported, messages, next_links = DoomedOPDSImporter(self._db).import_from_feed(feed)
-        
-        # One book was imported successfully.
-        [success] = imported
-        eq_("Johnny Crow's Party", success.title)
```

Comment:
this should still succeed, there's probably an exception in import_from_metadata that needs to be looked at.